gemma.cc

```diff
@@ -609,9 +697,14 @@ void GenerateImpl(GemmaImpl<TConfig>& gemma, size_t max_tokens,
   Activations<TConfig, 1>& activations = *gemma.state.get();
   Activations<TConfig, kPrefillBatchSize>& prefill_activations =
       *gemma.prefill.get();
-  const CompressedWeights<TConfig>& c_weights =
-      *reinterpret_cast<CompressedWeights<TConfig>*>(
-          gemma.compressed_weights.get());
+
+#if WEIGHTS_ARE_COMPRESSED
```

Comment:
How about adding near WEIGHTS_ARE_COMPRESSED a template typedef, something like
using <typename TConfig> WeightsT which is either CompressedWeights or Weights. Then here we do not require an #if.